pkg/distributor/query_ingest_storage_test.go

```diff
@@ -0,0 +1,378 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributor
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/grafana/dskit/grpcutil"
+	"github.com/grafana/dskit/test"
+	"github.com/grafana/dskit/user"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	ingester_client "github.com/grafana/mimir/pkg/ingester/client"
+	"github.com/grafana/mimir/pkg/mimirpb"
+	"github.com/grafana/mimir/pkg/querier/stats"
+	util_math "github.com/grafana/mimir/pkg/util/math"
+)
+
+func TestDistributor_QueryStream_ShouldSupportIngestStorage(t *testing.T) {
```

Comment:
do you want to add test cases where ingesters have different states in the ring (LEAVING, ACTIVE)? 




```go
	// ringStates explicitly sets the state of each ingester in the ring.
	// If unset, instances will be in ring.ACTIVE state.
	ringStates []ring.InstanceState
```

I had these on the partitions-ring branch, but I see you omitted them in this PR

Reply:
Thanks for spotting it. It wasn't intentional. I did remove all test cases querying both partitions and "classic" ring (which is not supported in this PR) and I unintentionally missed tests on ingesters state. Re-added in this commit: https://github.com/grafana/mimir/pull/7388/commits/a1d4757dbaa0aa52ddc9ea4f82370a865df261de